### PR TITLE
Add AppVeyor support.

### PR DESCRIPTION
**Pull request by XhmikosR**
This is the second attempt after a long time. Now AppVeyor is 10x faster so there shouldn't be a problem there.

Someone from the repo owners (/CC @parkr) should enable AppVeyor for the whole repo when the errors are fixed.

I tried to mimic the Travis config as close as possible.

BTW, if you want to see what is currently failing, check https://ci.appveyor.com/project/XhmikosR/jekyll until someone enables AppVeyor for this repo.

### CODE REVIEW

**Comment by XhmikosR** [comment]
BTW, if you want to see what is currently failing, check https://ci.appveyor.com/project/XhmikosR/jekyll until someone enables AppVeyor for this repo.

**Comment by parkr** [comment]
Thanks, @XhmikosR! Is there any configuration required on the AppVeyor side that I have to do in order to get this working?

**Comment by parkr** [comment]
LGTM. @jekyll/core, you down to add this? The tests will fail for a while but all the more reason to fix them. We have historically not supported Windows because we don't have development machines which run Windows but we could get closer to supporting it with this CI strategy.

/cc @jekyll/windows

**Comment by XhmikosR** [comment]
@parkr: only enable AppVeyor for the repo and perhaps enable "skip branches without appveyor.yml". Everything else should be set up already.

**Comment by XhmikosR** [comment]
@parkr:

>... and enable "skip branches without appveyor.yml"

**Comment by XhmikosR** [comment]
One thing that's different compared to Travis is

1. No jruby
2. fast_finish

For the first we can't do anything, but should I remove `fast_finish`?

**Comment by envygeeks** [comment]
@XhmikosR 

> No jruby

This is fine, our JRuby support is waiting for me (or really anybody) to get the time to finish cleaning it up so that @jekyll/core can make it official.  Right now it's lowdown supported simply because we use JRuby here where I am.

**Comment by XhmikosR** [comment]
@envygeeks: How about `fast_finish`?

**Comment by envygeeks** [comment]
@XhmikosR if we slim down what Rubies we support and can go parallel on Cucumber and Units I don't think that'll really be a big problem.  We use fast_finish on Travis because we've so many versions to test on. If @jekyll/core is cool I think for AppVeyor we only test on the most common version of Windows Ruby for now.

**Comment by envygeeks** [comment]
Let me rephrase this: "if we slim down what Rubies we support"
It should really read: "If we slim down what Rubies we test on."

**Comment by parkr** [comment]
> If @jekyll/core is cool I think for AppVeyor we only test on the most common version of Windows Ruby for now.

:+1: on this. No need for `fast_finish` or jruby on Windows.

**Comment by parkr** [comment]
LGTM.

**Comment by envygeeks** [comment]
LGTM.

**Comment by parkr** [comment]
@jekyllbot: merge +dev
